sfge/src/main/java/com/salesforce/rules/PerformNullCheckOnSoqlVariables.java

```diff
@@ -0,0 +1,149 @@
+package com.salesforce.rules;
+
+import com.google.common.collect.ImmutableSet;
+import com.salesforce.config.UserFacingMessages;
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.exception.UnexpectedException;
+import com.salesforce.graph.ApexPath;
+import com.salesforce.graph.source.ApexPathSource;
+import com.salesforce.graph.symbols.ScopeUtil;
+import com.salesforce.graph.symbols.SymbolProvider;
+import com.salesforce.graph.symbols.apex.ApexValue;
+import com.salesforce.graph.symbols.apex.Constraint;
+import com.salesforce.graph.vertex.BaseSFVertex;
+import com.salesforce.graph.vertex.SFVertex;
+import com.salesforce.graph.vertex.SoqlExpressionVertex;
+import com.salesforce.graph.vertex.VariableExpressionVertex;
+import java.util.*;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
+public final class PerformNullCheckOnSoqlVariables extends AbstractPathTraversalRule {
+
+    private static final ImmutableSet<ApexPathSource.Type> SOURCE_TYPES =
+            ImmutableSet.copyOf(ApexPathSource.Type.values());
+
+    private static final String URL =
+            "https://forcedotcom.github.io/sfdx-scanner/en/v3.x/salesforce-graph-engine/rules/#PerformNullCheckOnSoqlVariables";
+
+    // don't instantiate
+    private PerformNullCheckOnSoqlVariables() {}
+
+    @Override
+    public ImmutableSet<ApexPathSource.Type> getSourceTypes() {
+        return SOURCE_TYPES;
+    }
+
+    @Override
+    protected int getSeverity() {
+        return SEVERITY.MODERATE.code;
+    }
+
+    @Override
+    protected String getDescription() {
+        return UserFacingMessages.RuleDescriptions.PERFORM_NULL_CHECK_ON_SOQL_VARIABLE;
+    }
+
+    @Override
+    protected String getCategory() {
+        return CATEGORY.PERFORMANCE.name;
+    }
+
+    @Override
+    protected String getUrl() {
+        return URL;
+    }
+
+    public static PerformNullCheckOnSoqlVariables getInstance() {
+        return LazyHolder.INSTANCE;
+    }
+
+    @Override
+    protected boolean isEnabled() {
+        return true;
+    }
+
+    @Override
+    public boolean test(BaseSFVertex vertex) {
+        return false;
+    }
+
+    /**
+     * Note: this rule runs differently. Since the {@link SymbolProvider} is needed to determine
+     * whether a value might be null (and the symbol provider is unavailable during path walking and
+     * therefore the _run method), this rule only accepts vertices that are confirmed to be a
+     * violation. This is because the rule uses {@link Constraint}s to see if something has been
+     * null-checked, which are exclusively available during path expansion. Therefore, when walking
+     * the path, the rule automatically knows any accepted vertex is a violation.
+     */
```

Comment:
This doc tag is confusingly worded.
Can we get it into the more standard format?
```
/**
 * Short descriptor here.
 * @param vertex A Description
 * @param symbols Explanation of why we need this param, and its experimental nature
 */
```
You've basically got all of that information in the existing comment; it's mostly a matter of rearranging.